cpp-client/deephaven/client/include/private/deephaven/client/server/server.h

```diff
@@ -67,6 +67,63 @@ struct ServerResponseHolder final : public CompletionQueueCallback {
   Response response_;
 };
 
+// Options for the creation of the underlying grpc channel.
+class ClientOptions {
+public:
+  typedef std::vector<std::pair<std::string, int>> int_options_t;
+  typedef std::vector<std::pair<std::string, std::string>> string_options_t;
+public:
+  static ClientOptions defaults() {
+    return ClientOptions();
+  }
+
+  ClientOptions &setUseTls(const bool useTls) {
+    useTls_ = useTls;
+    return *this;
+  }
+
+  ClientOptions &setPem(const std::string &pem) {  // root certificate to use
+    pem_ = pem;
+    return *this;
+  }
+
+  // See https://grpc.github.io/grpc/cpp/group__grpc__arg__keys.html for
+  // a list of available options.
+  // Example:
+  //   copts.setStringOption("grpc.min_reconnect_backoff_ms", 2000)
+  ClientOptions &setIntOption(const std::string &opt, const int val) {
+    intOptions_.emplace_back(opt, val);
+    return *this;
+  }
+
+  // See https://grpc.github.io/grpc/cpp/group__grpc__arg__keys.html for
+  // a list of available options.
+  // Example:
+  //   copts.setStringOption("grpc.target_name_override", "idonthaveadnsforthishost");
+  ClientOptions &setStringOption(const std::string &opt, const std::string &val) {
+    stringOptions_.emplace_back(opt, val);
+    return *this;
+  }
```

Comment:
Prefer having doxygen comments, plus std::move style, as in
```
  /**
   * Sets an int-valued option.
   * See https://grpc.github.io/grpc/cpp/group__grpc__arg__keys.html for a list of available options.
   *
   * @example copt.setIntOption("grpc.min_reconnect_backoff_ms", 2000)
   * @param opt The option key.
   * @param val The option valiue.
   * @return *this, to be used for chaining
   */
  ClientOptions &setIntOption(std::string opt, const int val) {
    intOptions_.emplace_back(std::move(opt), val);
    return *this;
  }

  /**
   * Sets a string-valued option.
   * See https://grpc.github.io/grpc/cpp/group__grpc__arg__keys.html for a list of available options.
   *
   * @example copt.setStringOption("grpc.target_name_override", "idonthaveadnsforthishost")
   * @param opt The option key.
   * @param val The option valiue.
   * @return *this, to be used for chaining
   */
  ClientOptions &setStringOption(std::string opt, std::string val) {
    stringOptions_.emplace_back(std::move(opt), std::move(val));
    return *this;
  }
```

Reply:
Ok, I am going to add doxygen comments to ClientOptions.  But I reiterate, there are no doxygen comments anywhere else in this file.  createForTarget seems to want them, and I guess this file being in the private/ part of the dir structure is maybe not right?

Reply:
Part of what is happening is that my preferences are changing over time. I'm starting to believe in a lifestyle that says that that comments, when they exists, should be doxygen-style.

As to your public/private question. Now that I look at it, and assuming we are keeping the code in the same structure as it is now, then the only code that calls `Server::createFromTarget` is `Client::connect`.

Customers can only access `Client::connect` (because it is public). They don't even see Server::createFromTarget because it is private.

So this is starting to make me wonder whether your `ClientOptions` functionality (as defined in server.h) should be merged with my not-so-coincidentally-named `ClientOptions` class (defined in include/public/deephaven/client/client.h)

And then `Client::connect` can pull out the arguments it cares about (like it does now) and then pass the config object to `Server::createFromTarget`

I think to do this, `ClientOptions` should be moved to its own include file, in the public side of the directory tree, so that both Client and Server can see it.

Does this make sense? This comment is sort of complicated so we may need to discuss it offline.

Reply:
Furthermore, it would be the style of the rest of the code for the "private" code to never reference stuff on the "public side", so I think one way I *might* do it is to have the existing public-side ClientOptions have its own methods plus a pointer to an opaque ServerOptions class, and that ServerOptions class is what has your TLS stuff and is what is passed to Server::createFromTarget.

This is another thing which is hard to type but perhaps easy to explain in person. Or if you want me to add to the branch, I can show you what I mean, in code.